tests/utils.py

```diff
@@ -143,3 +145,19 @@ def run_pattern(pattern, data, metadata):
     reverse_transformed = pattern.reverse_transform(transformed)
 
     return updated_metadata, transformed, reverse_transformed
+
+
+def run_copula(data, metadata, patterns):
+    synthesizer = GaussianCopulaSynthesizer(metadata)
+    synthesizer.add_cag(patterns=patterns)
```

Comment:
also can we use the term constraint to replace pattern anywhere that is new code?

Reply:
@R-Palazzo already does this in his PR: https://github.com/sdv-dev/SDV/pull/2538/files.